tensorrt_llm/llmapi/llm_args.py

```diff
@@ -1210,7 +1169,7 @@ def _setup_speculative_config(self):
                         eagle3_one_model)
             elif isinstance(self.speculative_config, NGramDecodingConfig):
                 self.build_config.speculative_decoding_mode = SpeculativeDecodingMode.NGRAM
-                assert self.backend in ['pytorch', '_autodeploy']
+                assert self.backend == 'pytorch'
```

Comment:
I'm confused why this is if-else logic. Couldn't the respective speculative decoding configs handle their own validation and setup?

Reply:
Though that might require another option for defining which speculation mode you're using.

Reply:
It could work, but I just moved the original code here with minimum change. Just hope to keep the scope of this PR as it contains some bugfix and easy to get conflict.